roles/ngi_pipeline/templates/sourceme_common.sh.j2

```diff
@@ -27,3 +27,5 @@ export PS1="$IRMA_PROMPT [\u@\h \w]$ "
 
 # Default SNIC_TMP as /scratch
 export SNIC_TMP="/scratch"
+
+export NXF_SINGULARITY_CACHEDIR="{{ ngi_containers }}/biocontainers/nf-core-rnaseq_v3.0/"
```

Comment:
I wonder if this could interfere with other pipelines? Or will the `container ` parameter used for other pipelines take precedence over this?

Reply:
I think the older pipelines don't refer to this parameter yet. I checked with methylseq and it seems to run with `NXF_SINGULARITY_CACHEDIR` set. I'm guessing the other pipelines will be converted to use this parameter in the future. @ewels probably has a better idea about this?

Reply:
This is a core Nextflow option and so will affect all Nextflow pipelines: https://www.nextflow.io/docs/latest/config.html#environment-variables

It sets where the Singularity images are saved when they are downloaded from the internet. On a simpler system with an internet connection, you would just run any nf-core pipeline with `-profile singularity` and the images would automatically be downloaded to this folder.

Because we run offline, we can't download images, so we've never needed to set this. All of our existing pipelines refer to a _specific file path_ for their singularity containers - Nextflow doesn't download them and so no cachedir is used.

In the new `rnaseq` pipeline, we now have loads of containers instead of just one. Rather than having to manually set the file path for every container in the pipeline (would get boring and error prone quickly, especially with more pipelines), we let Nextflow do its thing with trying to download the images. However, we circumvent the problem of no internet connection by downloading the images to the cache directory ahead of time. So when Nextflow realises that it needs the image it looks there first, finds the image, and doesn't try to download it.

![simples](https://i.imgur.com/PB2VZ8U.png)

Reply:
TL;DR; - yes it will technically affect every Nextflow pipeline we have, but because they should all be specifying direct paths for the images it should not change the way that they run or present any problems.

I think.

Reply:
Cool cool! I guess it's pretty easy to test this by running another pipeline and see how it goes.